Add `Beatmap/Difficulty spread` (#9036)
ppy/osu-wiki: wiki/Beatmap/Difficulty_spread/en.md

---
tags:
  - difficulty set
  - progression
---

# Difficulty spread

A **difficulty spread** is a set of [difficulties](/wiki/Beatmap/Difficulty) sorted by [star rating](/wiki/Beatmap/Star_rating) in ascending order, such that:

> - There are no missing [difficulty levels](/wiki/Beatmap/Difficulty#difficulty-levels) — every difficulty between the easiest and the hardest ones from the spread's range is present.

Reviewer: technically you can have a spread with "missing" difficulties, basically you can omit any difficulty that's considered optional based on the drain time. for example, a 4:15 set that has easy-normal-insane-expert is rankable because the lowest diff required is insane, so you can omit any difficulty below that. (see [this](https://osu.ppy.sh/beatmapsets/1439897#taiko/2962574) and [this](https://osu.ppy.sh/beatmapsets/1452881#taiko/2987138) as example)

not sure if you want to cover this technicality or skip it.

Author: this is a valuable point, but I don't know how to insert into the existing sentence -- want to make up a suggestion?

Reviewer: thoughts?
```suggestion
- There are no missing [difficulty levels](/wiki/Beatmap/Difficulty#difficulty-levels) — every required difficulty between the easiest and the hardest ones from the spread's range is present.
   - Difficulties lower than the lowest difficulty level required, as determined by [spread rules](/wiki/Ranking_Criteria#rules.1), can be considered as optional and excluded from spread requirements.
```

- There are no significant difficulty gaps between two adjacent difficulties.

> The difficulty spread is a [ranked criteria](/wiki/Ranking_Criteria) requirement.

Reviewer: ```suggestion
The difficulty spread is a [ranking criteria](/wiki/Ranking_Criteria) requirement.
```

Author: wow

---
tags:
  - difficulty set
  - progression
---

# Difficulty spread

> A **difficulty spread** is a set of [difficulties](/wiki/Beatmap/Difficulty) sorted by [star rating](/wiki/Beatmap/Star_rating) in ascending order, such that:
> 
> - There are no missing [difficulty levels](/wiki/Beatmap/Difficulty#difficulty-levels) — every required difficulty between the easiest and the hardest ones from the spread's range is present.
>   - Difficulties lower than the lowest difficulty level required, as determined by [spread rules](/wiki/Ranking_Criteria#rules.1), can be considered as optional and excluded from spread requirements.
> - There are no significant difficulty gaps between two adjacent difficulties.
> 
> The difficulty spread is a [ranking criteria](/wiki/Ranking_Criteria) requirement.

Reviewer 2: the spread rules aren't requirements for something to be called a "difficulty spread"

```suggestion
A **difficulty spread** is a set of [difficulties](/wiki/Beatmap/Difficulty) in a [beatmap](/wiki/Beatmap) within some range of [star rating](/wiki/Beatmap/Star_rating) values.

[Ranked](/wiki/Beatmap/Category#ranked) beatmaps have several requirements regarding difficulty spread as per the [ranking criteria](/wiki/Ranking_Criteria):

- All required [difficulty levels](/wiki/Beatmap/Difficulty#difficulty-levels) within the spread's range must be present.
- Difficulties lower than the lowest difficulty level required, as determined by the [spread rules](/wiki/Ranking_Criteria#rules.1), are optional.
- No two adjacent beatmap difficulties may have a significant gap in difficulty.
```

Reviewer 2: don't know if summarising parts of the ranking criteria in this article is a good idea though

Author: well, that is essentially what a spread is, so it's fine by me
